tools/projmgr/docs/SystemArchitecture.md

```diff
@@ -0,0 +1,64 @@
+# CMSIS Project Manager - System Architecture [Draft]
+
+## Table of Contents
+
+[Introduction](#introduction)
+
+[Classes Overview](#classes-overview)
+
+[Data Flow](#data-flow)
+
+[Layers Combinations](#layers-combinations)
+
+
+## Introduction
+
+The CMSIS Project Manager is a C++ utility provided as binary and as library with interfaces for the most common programming languages and platforms.
+It leverages open source C++ libraries available in the [Open-CMSIS-Pack devtools](https://github.com/Open-CMSIS-Pack/devtools) repository.
+The tool assists the embedded software developer in the project creation by exposing available features
+from installed CMSIS Packs such as devices and components, allowing to search them using free text filters in addition to standard PDSC attributes. It also validates input files that are written in a
+human friendly YAML format following pre-defined schemas and it checks the correctness of components selection and unresolved missing dependencies.
+
+## Classes Overview
+
+The following diagram illustrates the relations among the project's C++ classes:
+
+![Classes](images/Classes.svg)
```

Comment:
I feel that it may be useful to have a relation with a cpackget "signature check module" allowing ProjMgr to verify a pack it is about to reference.
Optional of course.